src/main/java/uk/gov/hmcts/divorce/noticeofchange/service/ChangeOfRepresentativeService.java

```diff
@@ -96,12 +100,18 @@ private void setApplicantRepresented(Applicant applicant) {
         applicant.setOffline(YesOrNo.NO);
     }
 
-    private void updateOrgPolicyAndSolicitorDetails(Solicitor applicantSolicitor, String nocRequestUserFirmName,
+    private void updateOrgPolicyAndSolicitorDetails(Solicitor applicantSolicitor, OrganisationsResponse nocRequestUserOrg,
                                                     ProfessionalUser nocRequestingUser, String loggedInUserEmail) {
 
         applicantSolicitor.setName(String.join(" ", nocRequestingUser.getFirstName(), nocRequestingUser.getLastName()));
         applicantSolicitor.setEmail(loggedInUserEmail);
-        applicantSolicitor.setFirmName(nocRequestUserFirmName);
+        applicantSolicitor.setFirmName(nocRequestUserOrg.getName());
+        applicantSolicitor.setAddress(
+            parseOrganisationAddress(nocRequestUserOrg.getContactInformation())
+        );
+        applicantSolicitor.setReference(null);
+        applicantSolicitor.setAddressOverseas(null);
+        applicantSolicitor.setPhone(null);
```

Comment:
3 fields will be null:
- **Reference:** Seems to be an optional field that is not usually populated.
- **AddressOverseas:** We could possibly write a method that works this out from the address, but as the address won't be used for an online sol and so far with international post we have required that they explicitly specify when their address is international, it seems safest to leave it as null until the sol updates their own contact details.
- **Phone:** Doesn't seem to be stored by the professional API - the sol can fill this in later if they update their contact details.